### PR TITLE
[MOBILE] Handling of deleted data

### LINKED ISSUE #253: [BUG] Deleted user handling
- [x] Evaluate if we keep the user's data (app opinion, article being rent, etc...)
- [ ] Apply the choice previously made
- [ ] Adding corresponding tests for the handling of the user data

### PR DESCRIPTION
**Pull request by 31Nathan**
Now when a user is deleted every data is cleaned to lighten the database and to ensure no bug can appear due to residual data.
Closes #253

### CODE REVIEW

File: Server/controllers/Common/tickets.js [hunk 1]
```diff
@@ -94,3 +94,20 @@ exports.getConversation = (id) => {
...
+exports.cleanMobileUserTickets = (userId) => {
+  return db.Tickets.deleteMany({
+    where: {
+      OR: [
+        { creatorId: userId },
+        { assignedId: userId },
+      ]
+    }
+  });
+}
```
Comment (invpuppet) [inline Server/controllers/Common/tickets.js:113]: missing new line

File: Server/controllers/Mobile/cleandata.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * @param {*} userId of the user
+ * @param {*} notificationsId of the notifications linked to the user
+ */
+exports.cleanUserData = async (userId, notificationsId) => {
+    await favCtrl.cleanUserFavorite(userId);
+    await userCtrl.cleanUserNotifications(notificationsId);
+    await ticketCtrl.cleanMobileUserTickets(userId);
+    await rentCtrl.returnAllUserRents(userId);
+}
```
Comment (invpuppet) [inline Server/controllers/Mobile/cleandata.js:17]: missing new line

File: Server/controllers/Mobile/favorites.js [hunk 3]
```diff
@@ -103,3 +103,15 @@ exports.deleteFavorite = (favoriteId) => {
...
+ * Delete the user favorite before deleting the user
+ *
+ * @param {*} userId of the user
+ * @returns none
+ */
+exports.cleanUserFavorite = (userId) => {
+  return db.Favorite_Article_Mobile.deleteMany({
+    where: { userId: userId },
+  })
+}
```
Comment (invpuppet) [inline Server/controllers/Mobile/favorites.js:117]: missing new line

File: Server/controllers/Mobile/rent.js [hunk 4]
```diff
@@ -125,3 +125,20 @@ exports.updateRentInvoice = (rentId, invoiceData) => {
...
+exports.returnAllUserRents = async (userId) => {
+  rents = await this.getUserRents(userId)
+
+  rents.forEach(async element => {
+    await this.returnRent(element["id"])
+  });
+  return db.Location_Mobile.deleteMany({
+    where: { userId: userId },
+  })
+}
```
Comment (invpuppet) [inline Server/controllers/Mobile/rent.js:144]: missing new line

**Review by invpuppet — CHANGES_REQUESTED** [review]
missing tests and some new line

**Review by invpuppet — DISMISSED** [review]

**Review by invpuppet — APPROVED** [review]

**Review by CedricCORGE — APPROVED** [review]

**Pull request merged**